knimin/handlers/ag_search.py

```diff
@@ -58,5 +60,7 @@ def post(self):
             display_results.append(login_display)
 
         # now render the page
-        self.render("ag_search.html", results=display_results,
-                    handouts=handouts, currentuser=self.current_user)
+        self.render("ag_search.html",
+                    results=xhtml_escape_recursive(display_results),
+                    handouts=map(xhtml_escape, handouts),
```

Comment:
because the text might contain utf8 values which - I think - are rendered wrong if not escaped. But I'd be happy if you could double check.